### ISSUE TITLE
2024 updates

### ISSUE THREAD

**Issue by hannahker** [OCHA-DAP/pa-aa-cookiecutter#13]
This repo hasn't been updated in a while! Let's use this issue to track and brainstorm what we want to update with this cookiecutter: 

- [ ] Change prefix from `pa-` to `ds-` to reflect new team naming
- [ ] Add use of `.env` files for managing environment variables
- [ ] Replace existing linting and code formatting tools with `Ruff`? https://github.com/astral-sh/ruff

Some questions:
- Do we want to keep this specific to AA pilots? 
- What do we think about the directory structure? Is this still fit for purpose? 

```
├── analysis      # Main repository of analytical work for the AA pilot
├── docs          # .Rmd files or other relevant documentation
├── exploration   # Experimental work not intended to be replicated
├── src           # Code to run any relevant data acquisition/processing pipelines
|
├── .gitignore
├── README.md
└── requirements.txt
```

- Is it worth adding a flag to include boilerplate for any tests? GitHub actions?
- Would it be useful to include any standard data access functions? Eg. Reading files from the blob, accessing COGs, accessing Postgres database?

**Comment by t-downing** [comment]
> Do we want to keep this specific to AA pilots?

No. In general I think we need to be more flexible to non-framework AA and other types of analysis (this is a broader discussion, but I think this should be reflected in our code setup too). Perhaps here, this looks like just having the option of "AA framework?" yes/no. And if yes, include a couple additional things, like the status badge. 

> What do we think about the directory structure? Is this still fit for purpose?

More or less yes, although I would include more structure in `src`. Also something for pipeline scripts. Could be something like:
```
.
├── analysis
├── exploration
├── docs
├── pipelines/
│   ├── pipeline1.py
│   └── ...
└── src/
    ├── datasources/
    │   └── datasource1/
    │       ├── README.md
    │       └── [various sub-modules relevant to dataset, e.g. loading.py ...]
    └── utils/
        ├── blob.py
        └── [other utils, e.g. raster.py ...]
```
Not sure about where `blob.py` would best sit (and whether it should be called this), but this is typically what I've done. Also we could keep `datasource`s to a single `.py` file each (which is what I've always done), but this setup with individual `README`s does allow for nicer documentation. 

> Is it worth adding a flag to include boilerplate for any tests? GitHub actions?

I think I would leave this out for now, to keep things simple. But we can do it if you don't think it would complicate things.

> Would it be useful to include any standard data access functions? Eg. Reading files from the blob, accessing COGs, accessing Postgres database?

Love this! Would save us time and make it much quicker to jump into someone else's repo. I think at a basic level I'd want to put in:
- blob read/write
  - generic blob
  - `.parquet` blob
  - `.csv` blob
  - `.shp` blob - for this we may want local caching as well to save time
- blob open (i.e. lazy load)
  - COG
    - also, stack up COGs from a directory
  - netCDF?
- Postgres read
- some basic raster utilities? 

Maybe that's too long of a wishlist. But I think we all already have these utilities set up, so should be pretty easy to just put them into here.

**Comment by hannahker** [comment]
> No. In general I think we need to be more flexible to non-framework AA and other types of analysis (this is a broader discussion, but I think this should be reflected in our code setup too). 

Makes sense. However (and as you mentioned), I think we should use this more strictly as a tool for _analysis_ work. I think things like pipelines or apps are distinct enough that they will need their own setup. 

I'm curious to take a look at some existing analysis repos to see some examples of how people on the team are setting things up. @t-downing @zackarno @PaulineNimo are there any that you'd recommend taking a look at?

**Comment by t-downing** [comment]
> Makes sense. However (and as you mentioned), I think we should use this more strictly as a tool for analysis work. I think things like pipelines or apps are distinct enough that they will need their own setup.

Yup, sorry when I meant not just AA frameworks, I meant more broader types of _analysis_, as opposed to completely different types of code (like for apps). This cookiecutter was originally set up specifically for only AA frameworks, whereas I think it should be expanded to other, more ad hoc, analyses. Like I think overall our repos can be split into:
- analysis (I think we can generalize the cookiecutter to include all of this)
  - AA frameworks (very specific, and the only thing that the cookiecutter was originally intended for)
  - other AA-like analysis (e.g. Caribbean hurricanes monitoring, Cameroon flooding support)
  - other analysis (more random stuff, like Black Sea Grain Initiative)
- apps
- pipelines (however, many pipelines are currently inside repos that were mainly used for analysis. I think this ok, and actually makes sense particularly for AA frameworks where it's good to keep everything together)

Anyways I think generalizing would be easy- basically just need to: 

1. remove the "development"/"endorsed" badge since this isn't relevant if it's not a framework
2. remove `-aa-` from the repo name template
3. remove any other mention of AA

From my side some examples could be:

- [ds-aa-hti-hurricanes ](https://github.com/OCHA-DAP/ds-aa-hti-hurricanes) for structure of `src` and showing the monitoring `pipelines`, but I haven't yet distilled the `exploration` notebooks into `analysis` because of many relatively late-stage changes to the framework. So this is on my to-do
- [pa-aa-fji-storms](https://github.com/OCHA-DAP/pa-aa-fji-storms) for the split I was thinking of between `exploration` and `analysis` (although the `analysis` may be tricky to reproduce since it's GDrive based...), but honestly the `src` files in there are more of a mess and I'm not happy with the pipeline setup.
- [ds-cmr-flooding-support](https://github.com/OCHA-DAP/ds-cmr-flooding-support) for non-framework analyses (but pretty similar to the work we would do for an AA framework)
